lib/board.rb

```diff
@@ -0,0 +1,73 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    # resets the cells of the board to a 9 element array of ' '
+    reset!
+  end
+
+  def reset!
+    # reset the state of the cells in the board to a 9 element array of ' '
+    @cells = Array.new(9, ' ')
+  end
+
+  def display
+    # prints the board
+    puts "\n #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} \n\n"
+  end
+
+  def position(input)
+    # takes user input and returns the value of the board cell
+    cells[input.to_i-1]
+  end
+
+  def cells_filled?
+    # returns an array of the tokens that have been played
+    cells.find_all {|cell| cell == 'X' || cell == 'O'}
+  end
+
+  def turn_count
+    # returns the number of turns taken based on filled cells
+    cells_filled?.size
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```